### PR TITLE
Replace structopt with its successor, clap 3

### PR DESCRIPTION
**Pull request by Ralith**
Only affects examples/tools.

### CODE REVIEW

**Comment by Ralith** [comment]
Looks like this is blocked on updating MSRV to 1.54.

**Comment by djc** [comment]
See https://github.com/clap-rs/clap/issues/3267.

File: bench/Cargo.toml [hunk 1]
```diff
@@ -12,7 +12,7 @@ hdrhistogram = { version = "7.2", default-features = false }
 quinn = { path = "../quinn" }
 rcgen = "0.8"
 rustls = { version = "0.20", default-features = false, features = ["quic"] }
-structopt = "0.3"
+clap = { version = "3.0.10", features = ["derive"] }
```
Comment (KirillLykov) [inline bench/Cargo.toml:15]: Why did you choose `3.0.10` instead of `3.1.x`?

File: perf/src/bin/perf_client.rs [hunk 2]
```diff
@@ -16,54 +16,54 @@ use perf::stats::{OpenStreamStats, Stats};
...
 
 /// Connects to a QUIC perf server and maintains a specified pattern of requests until interrupted
-#[derive(StructOpt)]
-#[structopt(name = "client")]
+#[derive(Parser)]
+#[clap(name = "client")]
 struct Opt {
     /// Host to connect to
-    #[structopt(default_value = "localhost:4433")]
+    #[clap(default_value = "localhost:4433")]
```
Comment (KirillLykov) [inline perf/src/bin/perf_client.rs:23]: Looks like the only place we you use string instead of `SocketAddr`, might be good to format in later prs

Comment (Ralith) [reply perf/src/bin/perf_client.rs:23]: `SocketAddr` cannot represent hostnames.

**Comment by Ralith** [comment]
This should be mergeable now that we've bumped MSRV.

**Review by djc — APPROVED** [review]

**Pull request merged**